frontend/src/app/charts/charts.component.ts

```diff
@@ -84,29 +77,36 @@ export class ChartsComponent implements OnInit {
     this.route.params.forEach((params: Params) => {
       this.repoName = params['repo'] ? params['repo'] : undefined;
       this.updateMetaTags();
-      this.loadRepos();
       this.loadCharts();
     });
   }
 
   loadCharts(): void {
-    this.chartsService.getCharts(this.repoName).subscribe(charts => {
+    this.chartsService.getCharts().subscribe(allCharts => {
       this.loading = false;
-      this.charts = charts;
+      this.charts = allCharts.filter(c => !this.repoName || c.attributes.repo.name === this.repoName);
       if (!this.searchTerm) {
         this.orderedCharts = this.orderCharts(this.charts);
       }
+      this.setReposFromCharts(allCharts);
     });
   }
-
-  loadRepos(): void {
-    this.reposService.getRepos().subscribe(repos => {
-      repos.splice(0, 0, this.allRepo);
-      this.filters[0].items = repos.map(r => ({
-        title: r.attributes.name,
-        value: r.id,
-        selected: this.repoName ? r.id == this.repoName : r.id == 'all'
-      }));
+  
+  setReposFromCharts(charts: Chart[]): void {
```

Comment:
It looks good. I'd just add a comment in the code explaining what this does and why it is needed.

Thanks!